### ISSUE TITLE
How to best support environment specific overrides

### ISSUE THREAD

**Issue by berstend** [kevincharm/parcel-plugin-web-extension#11]
Hey there,

while working on a non-trivial chrome extension I came across the need to adjust the `manifest.json` based on the environment.

More specifically to enable HMR `unsafe-eval` must be allowed in the [CSP](https://developer.chrome.com/extensions/contentSecurityPolicy#relaxing-eval) configuration (in conjunction with `parcel watch --hmr-hostname localhost`).

As `unsafe-eval` is pretty evil I'd like to only use that during development, not in the production build.

I'm wondering if we should add support for environment specific configs or if that's outside of the scope of this plugin. :-)

Right now I can just switch between manifests based on the environment:
```bash
export NODE_ENV=development && parcel watch src/manifest.${NODE_ENV}.json -d build/${NODE_ENV}/
```
This results in a lot of duplicated info, as I have to maintain two full manifests just for a single entry that's different.

One potential solution could be to additionally support optional `manifest.development.json` and `manifest.production.json` files, that include specific overrides:

```json
// manifest.development.json:
{
  "content_security_policy": "script-src 'self' 'unsafe-eval'; object-src 'self'"
}
```

If one of these files exists they could be automatically[ deep-merged](https://github.com/berstend/puppeteer-extra/blob/master/packages/puppeteer-extra-plugin/index.js#L53-L56) (depending on `NODE_ENV`) with the base file.

Alternatively we could support `manifest.json.js` files, that just return a manifest JSON as default export, after compiling it's contents arbitrarily.

Another use-case where environment specific config would come in handy is e.g. [externally_connectable](https://developer.chrome.com/extensions/manifest/externally_connectable).

What are your thoughts on this matter? :-)

Thanks!

**Comment by kevincharm** [comment]
Hey again! Your `manifest.${env}.json` approach looks pretty good for that problem. Similar to the idiomatic way of doing dev/prod webpack configs. We should also be able to infer the environment from the cli/build options.
I'll have a look around and see what other Parcel plugins are doing first.

**Comment by kevincharm** [comment]
Hey @berstend, please check out #15 for an initial implementation. It doesn't do a deep merge (just replaces top level keys)... let me know if that feels like a good solution.

**Comment by kevincharm** [comment]
Merged #15, released in v1.4.0.

**Comment by berstend** [comment]
Awesome @kevincharm! top level key merging seems perfectly fine for most use-cases. Great job, thanks! 👍
